src/submit.rs

```diff
@@ -275,16 +275,14 @@ impl<'a> Submitter<'a> {
     /// credentials of the task that originally registered the io_uring.
     ///
     /// [`Parameters::is_feature_cur_personality`]: crate::Parameters::is_feature_cur_personality
-    pub fn register_personality(&self) -> io::Result<Personality> {
+    pub fn register_personality(&self) -> io::Result<u16> {
         let id = execute(
             self.fd.as_raw_fd(),
             sys::IORING_REGISTER_PERSONALITY,
             ptr::null(),
             0,
         )?;
-        Ok(Personality {
-            id: id.try_into().unwrap(),
-        })
+        Ok(id.try_into().unwrap())
```

Comment:
I think we can avoid `unwrap`, because `execute` has already determined that` ret >= 0`.

maybe we should let `execute` return u16.

Reply:
Would you prefer I just use `id as u16` here?

Reply:
I believe it makes more sense to make `execute` return u16.

Reply:
Is there the possibility that `execute` returns an i32 that doesn't fit in a u16?

Reply:
Currently `execute` never return negative number. If there is a new method that needs to return i32 in future, we can also simply change it without breaking API.

Reply:
Ok, but since `io_uring_register` returns an i32, isn't it possible it could be a positive number larger than `u16::MAX` (65535)? (Sorry if I'm not understanding something here)

Reply:
You're right... It's four o'clock in the morning, I might need sleep.

I think `id as u16` is good.

Reply:
Ok, no problem at all! :)